src/ui/Base/ComponentOptions.ts

```diff
@@ -657,16 +657,18 @@ export class ComponentOptions {
   }
 
   static findParentScrolling(element: HTMLElement, doc: Document = document): HTMLElement {
-    while (<Node>element != doc && element != null) {
-      if (ComponentOptions.isElementScrollable(element)) {
-        if (element.tagName.toLowerCase() !== 'body') {
-          return element;
-        }
-        return <any>window;
-      }
-      element = element.parentElement;
+    element = this.findParentScrollLockable(element, doc);
+    return element instanceof HTMLBodyElement || !ComponentOptions.isElementScrollable(element) ? <any>window : element;
+  }
+
+  static findParentScrollLockable(element: HTMLElement, doc: Document = document): HTMLElement {
+    if (!element) {
+      element = doc.body;
+    }
+    if (ComponentOptions.isElementScrollable(element) || element instanceof HTMLBodyElement || !element.parentElement) {
+      return element;
     }
-    return <any>window;
+    return ComponentOptions.findParentScrollLockable(element.parentElement, doc);
   }
```

Comment:
Do you believe this could be breaking? Should I revert `findParentScrolling`?

Reply:
Hmm that's an old method 😅 Was there an issue with the previous implementation? 
I don't see the UTs breaking. but then again the UTs were not very complete. I'd suggest thoroughly testing the possibles paths for `findParentScrolling` & the new `findParentScrollLockable` methods (or make `findParentScrollLockable` private). 
Might also be useful to first write the tests for the older implementation and then see that your implementation is not having different results.

Reply:
There was no issue with the previous implementation. However, I needed a version that would always return a style-able element (`window` isn't style-able). I created that version and ended up with `findParentScrollLockable`. However. a lot of code from `findParentScrollLockable` seemed reusable for `findParentScrolling`.

Since I need to call `findParentScrollLockable`, I can't make it private.

Good idea, I'll add some more tests.